tests/test_manifest.py

```
@@ -0,0 +1,132 @@
+            self.assertFalse(
+                MUTABLE_NPM_REF.search(token),
+                f"mutable npm reference in manifest command: {token}",
+            )
+
+    def test_lockfile_pins_expected_tarball_and_integrity(self) -> None:
+        self.assertTrue(self.package["private"])
+        self.assertEqual(
+            self.package["dependencies"]["ccusage"], CCUSAGE_VERSION
+        )
+        root = self.lock["packages"][""]
+        package = self.lock["packages"][f"node_modules/ccusage"]
```

Ruff (0.16.3)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)